Program.cs

```diff
@@ -1,2 +1,70 @@
-﻿// See https://aka.ms/new-console-template for more information
-Console.WriteLine ("Hello, World!");
+﻿// Test Cases
+MyStack<int> stack = new ();
+stack.Push (1);
+stack.Push (2);
+stack.Push (3);
+stack.Push (4);
+stack.Push (5);
+stack.Display ();
+Console.WriteLine (stack.Pop ());
+stack.Display ();
+stack.Push (6);
+stack.Display ();
+Console.WriteLine (stack.Peek ());
+
+/// <summary> The class MyStack defines the implementation of stack with array as the underlying data structure </summary>
+/// <typeparam name="T">Datatype of the elements in the stack</typeparam>
+class MyStack<T> {
+   T[] mArray;
+   int mTop;
+
+   /// <summary> Constructor of MyStack class </summary>
+   public MyStack () {
+      mArray = new T[4];
+      mTop = -1;
+   }
+
+   /// <summary> Property with top variable </summary>
+   public int Top => mTop;
+
+   /// <summary> Property with the capacity of the stack </summary>
+   public int Capacity => mArray.Length;
+
+   /// <summary> Push method for adding(pushing) elements into the stack </summary>
+   /// <param name="a">the element to be pushed</param>
+   public void Push (T a) {
+      if (mTop == (Capacity - 1)) Array.Resize (ref mArray, Capacity * 2);
+      mTop++;
+      mArray[mTop] = a;
+   }
+
+   /// <summary> Pop method for deleting(popping) the last element from the stack </summary>
+   /// <returns> Returns the array after popping </returns>
+   /// <exception cref="InvalidOperationException"></exception>
+   public T Pop () {
+      if (IsEmpty) throw new InvalidOperationException ();
+      var item = mArray[mTop];
+      Array.Clear (mArray, mTop, 1);
+      mTop--;
+      return item;
+   }
+
+   /// <summary> Peek method for returning the last element of the stack </summary>
+   /// <returns> Returns the last element </returns>
+   /// <exception cref="InvalidOperationException"></exception>
+   public T Peek () {
+      if (IsEmpty) throw new InvalidOperationException ();
+      return mArray[mTop];
+   }
+
+   /// <summary> IsEmpty property to check whether the stack is empty </summary>
+   public bool IsEmpty => (mTop == -1) ? true : false;

```

Comment:
No need for if else if the branch result is true and false. We could just return the condition of the if condition itself
```suggestion
   public bool IsEmpty => mTop == -1;

```